nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/CopyS3Object.java

```diff
@@ -119,28 +145,127 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         final String destinationBucket = context.getProperty(DESTINATION_BUCKET).evaluateAttributeExpressions(flowFile).getValue();
         final String destinationKey = context.getProperty(DESTINATION_KEY).evaluateAttributeExpressions(flowFile).getValue();
 
+        final GetObjectMetadataRequest sourceMetadataRequest = new GetObjectMetadataRequest(sourceBucket, sourceKey);
+        final ObjectMetadata metadataResult = s3.getObjectMetadata(sourceMetadataRequest);
+
+        if (metadataResult == null) {
+            session.transfer(flowFile, REL_FAILURE);
+        }
```

Comment:
I recommend removing this check, since it could be handled through the the GetObjectMetadata Processor. This introduces another request which may not be necessary in many cases.

Reply:
I have mixed feelings about that because V1 of the Java AWS S3 API can return a null instead of throwing an exception. I'm not sure I like the idea of relying on an upstream processor.